src/pools/D3MAavePool.sol

```diff
@@ -173,8 +181,10 @@ contract D3MAavePool is ID3MPool {
         require(dai.balanceOf(msg.sender) == prevDai + wad, "D3MAavePool/incorrect-dai-balance-received");
     }
 
-    function transfer(address dst, uint256 wad) external override onlyHub {
-        require(adai.transfer(dst, wad), "D3MAavePool/transfer-failed");
+    function exit(address dst, uint256 wad) external override onlyHub {
+        uint256 amt = wad * assetBalance() / (D3mHubLike(hub).end().Art(ilk) - exited);
```

Comment:
Logic looks good.

If we care much about gas here we can:
* Consider keeping `end` in storage (fileable).
* Caching `exited` in the beginning and doing something like:  `exited = _exited +  wad`

Reply:
- I started with a fileable `end` but then when I realized it could come from the `hub` change to the actual one. I know is more gas intensive but at the same time will save us complication for initial set up and in case of `End` migration.
Anyway not that strong feeling if you prefer the other way.
- Done

Reply:
Ok, I'm also fine with the way `end` is fetched from the hub.
Resolving.